freqtrade/freqtradebot.py

```diff
@@ -192,7 +199,11 @@ def process(self) -> None:
 
         strategy_safe_wrapper(self.strategy.bot_loop_start, supress_error=True)()
 
-        self.strategy.analyze(self.active_pair_whitelist)
+        # This just means we won't broadcast dataframes if we're listening to a producer
```

Comment:
:+1: Maybe we should just base it on `enable_message_ws` in the config for if the bot should be broadcasting here, regardless of if it's using any consumers

Reply:
:+1:

Reply:
So I think that might be a little harder than we'd think, as doing something like `self.config.get('api_server', {}).get('enable_message_ws', False)` could make us emit the dataframe if that stanza in the configuration is defined, but not even enabled. Checking with the dataprovider means if we are supposed to be listening to producers then don't broadcast, if we aren't listening producers, then do broadcast the dataframe.

Reply:
what would be the usecase for the following scenario:

`leader --> <broadcasts> --> follower --> <broadcasts> --> follower 2`

sure, you could build "chains" of followers, with each adding one column ... but what's the real world usecase for this?

if you're bad at configuring things (or unlucky), you can have one bot send the data to itself as it connects to it's own websocket ... that'd probably not be wanted, and will be quite difficult to debug unless you really know what you're doing and got access to that system (which we usually won't).

---

on the other hand - what's bad with  "attempting" to broadcast again? we can always call `send_msg()` - if no one is listening, no one would get the message. 
if the webserver is down - there'll be no handler for that message - so it's just 2-3 calls that go into the void.

Reply:
I see no reason to add that. Unless otherwise changed in the future, having the follower broadcast it's own dataframes just won't happen. The only time we don't want to "attempt" to broadcast is if we are a follower, as that would waste time serializing dataframes just to try and send to no one.

Reply:
> as that would waste time serializing dataframes just to try and send to no one.

i think the serialization should only happen once we're at the "found sender - sending" step.
that'll avoid serialization at all if noone is listening (which at the beginning, will probably be like 99% of all running bots).

Reply:
This is fixed, check the recent commit ([dccde88](https://github.com/freqtrade/freqtrade/pull/7303/commits/dccde88c83c4e8dfa7c8b70a6c4a7f9459ce177c))